components/script/dom/bindings/structuredclone.rs

```diff
@@ -52,7 +57,36 @@ pub(super) enum StructuredCloneTags {
     Max = 0xFFFFFFFF,
 }
 
-unsafe fn read_blob(
+impl From<SerializableInterface> for StructuredCloneTags {
+    fn from(v: SerializableInterface) -> Self {
+        match v {
+            SerializableInterface::Blob => StructuredCloneTags::DomBlob,
+        }
+    }
+}
+
+impl From<TransferrableInterface> for StructuredCloneTags {
+    fn from(v: TransferrableInterface) -> Self {
+        match v {
+            TransferrableInterface::MessagePort => StructuredCloneTags::MessagePort,
+        }
+    }
+}
+
+fn reader_for_type(
+    val: SerializableInterface,
```

Comment:
Can you not pass `StructuredCloneTags` here and match in it, it seems this would enable removing the loop with the `StructuredCloneTags::from(serializable)` check, but it's hard to tell without trying.

Reply:
I don't believe this is possible in this design. StructuredCloneTags includes all serializable and transferable types plus additional marker values, which is why I wanted a separate enum with only the relevant interfaces. I _think_ what you're asking for is "given this structured clone tag value, give me the enum value corresponding to the type we need to deserialize", but that requires a match expression which is much harder to make impossible to get wrong.